gmso/formats/lammpsdata.py

```diff
@@ -654,13 +719,437 @@ def _get_ff_information(filename, unit_style, topology):
+        out_file.write(
+            "{:d} angle types\n".format(len(top.angle_types(filter_by=pfilter)))
+        )
+    if top.n_dihedrals > 0 and atom_style in ["full", "molecular"]:
+        out_file.write(
+            "{:d} dihedral types\n".format(
+                len(top.dihedral_types(filter_by=pfilter))
+            )
+        )
+    if top.n_impropers > 0 and atom_style in ["full", "molecular"]:
+        out_file.write(
+            "{:d} improper types\n".format(
+                len(top.improper_types(filter_by=pfilter))
+            )
+        )
+
+    out_file.write("\n")
+
+
+def _write_box(out_file, top, base_unyts, cfactorsDict):
+    """Write GMSO Topology box to LAMMPS file."""
+    # TODO: unit conversions
+    if allclose_units(
+        top.box.angles,
+        u.unyt_array([90, 90, 90], "degree"),
+        rtol=1e-5,
+        atol=1e-8,
+    ):
+        box_lengths = [
+            _parameter_converted_to_float(
+                top.box.lengths[i], base_unyts, cfactorsDict
+            )
+            for i in range(3)
+        ]
+        for i, dim in enumerate(["x", "y", "z"]):
+            out_file.write(
+                "{0:.6f} {1:.6f} {2}lo {2}hi\n".format(0, box_lengths[i], dim)
+            )
+        out_file.write("0.000000 0.000000 0.000000 xy xz yz\n")
+    else:
+        top.box.lengths.convert_to_units(u.angstrom)
+        top.box.angles.convert_to_units(u.radian)
+        vectors = top.box.get_vectors()
+
+        xhi = vectors[0][0]
+        yhi = vectors[1][1]
+        zhi = vectors[2][2]
+        xy = vectors[1][0]
+        xz = vectors[2][0]
+        yz = vectors[2][1]
+        xlo = u.unyt_array(0, xy.units)
+        ylo = u.unyt_array(0, xy.units)
+        zlo = u.unyt_array(0, xy.units)
+
+        xlo_bound = xlo + u.unyt_array(np.min([0.0, xy, xz, xy + xz]), xy.units)
+        xhi_bound = xhi + u.unyt_array(np.max([0.0, xy, xz, xy + xz]), xy.units)
+        ylo_bound = ylo + u.unyt_array(np.min([0.0, yz]), xy.units)
+        yhi_bound = yhi + u.unyt_array(np.max([0.0, yz]), xy.units)
+        zlo_bound = zlo
+        zhi_bound = zhi
+
+        out_file.write(
+            "{0:.6f} {1:.6f} xlo xhi\n".format(xlo_bound.value, xhi_bound.value)
+        )
+        out_file.write(
+            "{0:.6f} {1:.6f} ylo yhi\n".format(ylo_bound.value, yhi_bound.value)
+        )
+        out_file.write(
+            "{0:.6f} {1:.6f} zlo zhi\n".format(zlo_bound.value, zhi_bound.value)
+        )
+        out_file.write(
+            "{0:.6f} {1:.6f} {2:.6f} xy xz yz\n".format(
+                xy.value, xz.value, yz.value
+            )
+        )
+
+
+def _write_atomtypes(out_file, top, base_unyts, cfactorsDict):
+    """Write out atomtypes in GMSO topology to LAMMPS file."""
+    # TODO: Get a dictionary of indices and atom types
+    # TODO: Allow for unit conversions for the unit styles
+    out_file.write("\nMasses\n")
+    out_file.write(f"#\tmass ({top.sites[0].mass.units})\n")
```

Comment:
yes, all of these should use the base_unyt value now that the conversion is happening during the writing step. However, this will get changes with the upcoming unit.py refactoring.